src/error.rs

```diff
@@ -0,0 +1,7 @@
+use std::time::Duration;
+
+#[derive(Debug, PartialEq, Eq)]
+pub enum Error {
```

Comment:
Shall this enum implement an `Error` trait?

Reply:
Yeah, good idea!